server/planning/planning/planning_featured_async_service.py

```diff
@@ -0,0 +1,108 @@
+from datetime import datetime
+from typing import Any
+from copy import deepcopy
+
+from apps.auth import get_user_id
+from bson import ObjectId
+from planning.core.service import BasePlanningAsyncService
+from planning.common import (
+    get_version_item_for_post,
+    enqueue_planning_item,
+)
+from planning.planning.planning_service import PlanningAsyncService
+from planning.types import PlanningFeaturedResourceModel
+from superdesk import get_resource_service, logger
+from superdesk.core import get_app_config
+from superdesk.core.types import SearchRequest
+from superdesk.errors import SuperdeskApiError
+from superdesk.utc import utc_to_local, utcnow
+
+ID_DATE_FORMAT = "%Y%m%d"
+
+
+class PlanningFeaturedAsyncService(BasePlanningAsyncService[PlanningFeaturedResourceModel]):
+    """Async Service class for the planning featured model."""
+
+    async def on_create(self, docs: list[PlanningFeaturedResourceModel]):
+        await super().on_create(docs)
+
+        for doc in docs:
+            date = utc_to_local(doc.tz or get_app_config("DEFAULT_TIMEZONE"), doc.date)
+            _id = date.strftime(ID_DATE_FORMAT)
+
+            search_request = SearchRequest(where={"_id": _id})
+            items = await super().find(search_request)
+            if await items.count() > 0:
+                raise SuperdeskApiError.badRequestError(message="Featured story already exists for this date.")
+
+            await self.validate_featured_attrribute(doc.items)
+            doc.id = _id
+            await self.post_featured_planning(doc)
+
+    async def on_created(self, docs: list[PlanningFeaturedResourceModel]):
+        for doc in docs:
+            await self.enqueue_published_item(doc, doc)
+
+    async def on_update(self, updates: dict[str, Any], original: PlanningFeaturedResourceModel):
+        await super().on_update(updates, original)
+        # Find all planning items in the list
+        added_featured = [item_id for item_id in updates.get("items") or [] if item_id not in original.items or []]
+        await self.validate_featured_attrribute(added_featured)
+        await self.post_featured_planning(PlanningFeaturedResourceModel(**updates), original)
+
+    async def on_updated(self, updates: dict[str, Any], original: PlanningFeaturedResourceModel):
+        await self.enqueue_published_item(PlanningFeaturedResourceModel(**updates), original)
```

Comment:
Missing calls to super class' events. Such as `on_created`, `on_updated` etc. It's important to call super methods in case a parent class performs some actions.

Such as [resource signals](https://github.com/superdesk/superdesk-core/blob/async/superdesk/core/resources/service.py#L487)